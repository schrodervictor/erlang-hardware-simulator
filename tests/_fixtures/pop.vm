pop local 3
pop argument 5
pop this 42
pop that 10
pop static 34
pop temp 8
pop pointer 1
pop pointer 0
