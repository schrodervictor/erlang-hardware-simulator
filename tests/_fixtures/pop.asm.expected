// pop local 3
@LCL
D=M
@3
D=D+A
@addr
M=D
@SP
AM=M-1
D=M
@addr
A=M
M=D
// pop argument 5
@ARG
D=M
@5
D=D+A
@addr
M=D
@SP
AM=M-1
D=M
@addr
A=M
M=D
// pop this 42
@THIS
D=M
@42
D=D+A
@addr
M=D
@SP
AM=M-1
D=M
@addr
A=M
M=D
// pop that 10
@THAT
D=M
@10
D=D+A
@addr
M=D
@SP
AM=M-1
D=M
@addr
A=M
M=D
// pop static 34
@SP
AM=M-1
D=M
@pop.34
M=D
// pop temp 8
@SP
AM=M-1
D=M
@13
M=D
// pop pointer 1
@SP
AM=M-1
D=M
@THAT
M=D
// pop pointer 0
@SP
AM=M-1
D=M
@THIS
M=D
